### ISSUE TITLE
Component model

### ISSUE THREAD

**Issue by anandijain** [SciML/CellMLToolkit.jl#39]
Also a test on component model will be great to have

**Comment by shahriariravanian** [comment]
Look at https://libcellml.org/ . This is the official C++ library from the CellML folks. It is still a work in progress, but has many interesting test scenarios (components included). Let's take a look and see if we can adapt the tests.

**Comment by anandijain** [comment]
https://github.com/cellml/libcellml/issues/838 

We should team up with these guys to come up with the DAE library

**Comment by ChrisRackauckas** [comment]
Note that you can use `ODAEProblem` and `DAEProblem`, instead of just `ODEProblem`, to handle alternative lowering behaviors like having `D(x)` on the RHS.

**Comment by shahriariravanian** [comment]
I agree. I've been going through the files that give an error and try to find the reason, especially for "value of X is not found" errors. Some CellML files are just defective (missing the initial_value), but some are caused by the circular definition of algebraic variables, which need DAE to solve.

**Comment by anandijain** [comment]
[Feto](https://models.cellml.org/e/4e2/Feto_maternal_Oxygen_transfer.cellml/license_citation) is an example of the circular definition, right?

What's a model missing u0?

**Comment by shahriariravanian** [comment]
Probably, but it is such a large model that I haven't got to it. I'm starting with smaller models. One example is "saucerman_mcculloch_2004/saucerman_mcculloch_2004.cellml" (variable PKACII_RyR). Another example is "siebert_rode_herzog_till_blickhan_2008/siebert_CC_2008.cellml" (variables f_v and v_cc).
